src/databases/Mili/avtMiliFileFormat.C

```diff
@@ -74,4384 +53,3837 @@ extern "C" {
+         break;
+       }
+       case M_FLOAT8:
+       {
+           dTmp = (double *) readBuff;
+           for (int i = 0 ; i < dataSize ; i++)
+           {
+               dataBuff[i] = (float)(dTmp[i]);
+           }
+           delete [] dTmp;
+           break;
+       }
+       case M_INT:
+       {
+            //
+            // Intentional fallthrough.
+            //
+       }
+       case M_INT4:
+       {
+         iTmp = (int *) readBuff;
+         for (int i = 0 ; i < dataSize ; i++)
+         {
+             dataBuff[i] = (float)(iTmp[i]);
+         }
+         delete [] iTmp;
+         break;
+       }
+       case M_INT8:
+       {
+           lTmp = (long *) readBuff;
+           for (int i = 0 ; i < dataSize ; i++)
+           {
+               dataBuff[i] = (float)(lTmp[i]);
+           }
+           delete [] lTmp;
+           break;
+       }
+       default:
+       {
+           break;
+       }
     }
 }
 
+
 // ****************************************************************************
 //  Constructor:  avtMiliFileFormat::avtMiliFileFormat
 //
+//  This method was re-written from its original version. 
+//
 //  Arguments:
-//    fname      the file name of one of the Mili files.
+//    fpath      The path to a .mili json file. 
 //
-//  Programmer:  Hank Childs
-//  Creation:    April 11, 2003
+//  Programmer:  Alister Maguire
+//  Creation:    Jan 15, 2019
 //
 //  Modifications
-//    Akira Haddox, Fri May 23 08:13:09 PDT 2003
-//    Added in support for multiple meshes. Changed to MTMD.
-//    Changed to read in .mili format files.
-//
-//    Akira Haddox, Tue Jul 22 15:34:40 PDT 2003
-//    Initialized setTimesteps.
-//
-//    Akira Haddox, Fri Jul 25 11:09:13 PDT 2003
-//    Added reading in of variable dimensions.
-//
-//    Akira Haddox, Mon Aug 18 14:33:15 PDT 2003
-//    Added partition file support.
-//
-//    Hank Childs, Tue Jul 20 15:53:30 PDT 2004
-//    Add support for more data types (float, double, char, int, etc).
-//
-//    Mark C. Miller, Mon Jul 18 13:41:13 PDT 2005
-//    Added initialization of structures having to do with free nodes mesh
-//
-//    Kathleen Bonnell, Wed Aug 5 17:44:22 PDT 2009 
-//    Test for windows-style slashes when scanning filename.
 //     
 // ****************************************************************************
 
-avtMiliFileFormat::avtMiliFileFormat(const char *fname)
-    : avtMTMDFileFormat(fname)
+avtMiliFileFormat::avtMiliFileFormat(const char *fpath)
+    : avtMTMDFileFormat(fpath)
 {
+    dims       = 0;
+    nDomains   = 0;
+    nMeshes    = 0;
+    nTimesteps = 0;
+    datasets   = NULL;
+    materials  = NULL;
 
-#ifdef MILI_VERSION
-    if ( !miliLoadMessageDisplayed ) {
-         printf("\nUsing MDG Visit Mili Plugin. Visit Version %s / Mili Version %s (%s)\n\n\n", VISIT_VERSION, MILI_VERSION, DATE_VERSION );
-    }
-#endif
+    LoadMiliInfoJson(fpath);
+
+    string fnamePth(fpath);
+
+    //TODO: make sure this works for windows paths as well. 
```

Comment:
Did you satisfy this todo?

Reply:
Yup, I'll go ahead and remove that TODO.